### PR TITLE
Improving MODX internal messaging between users

### PR DESCRIPTION
**Pull request by alroniks**
### What does it do?
This improvement adds an ability to see received and sent messages. Also changed the order of fields because of a subject the mosts important thing in the message list. Also, fullname of user showed in the message list for best conveniences.

### Why is it needed?
Actually, a user can send messages and see messages that were sent to him. Unfortunately, the user can not look at messages that he sent. 

Video about how it works:
https://www.youtube.com/watch?v=0mkLBuClTIg

### Related issue(s)/PR(s)
Not found, but please, attach if you know.

### CODE REVIEW

**Comment by sottwell** [comment]
Just for information... I have a Dashboard widget to interact with messages https://gist.github.com/sottwell/bca3dc4d2de4b15a5261

**Comment by Mark-H** [comment]
While I wouldn't mind to see user messaging taken out (extracted into an extra) in 3.0, this looks like a good improvement that might actually make it useful ;) Thanks @Alroniks!

File: core/model/modx/processors/security/message/getlist.class.php [hunk 1]
```diff
@@ -1,58 +1,97 @@
...
+    public function prepareQueryBeforeCount(xPDOQuery $c)
+    {
+        $c->innerJoin('modUser', 'Sender');
+        $c->innerJoin('modUser', 'Recipient');
+        $c->innerJoin('modUserProfile', 'SenderProfile', 'SenderProfile.id = modUserMessage.sender');
+        $c->innerJoin('modUserProfile', 'RecipientProfile', 'RecipientProfile.id = modUserMessage.recipient');
+
+        switch ($this->getProperty('type')) {
+            case 'outbox':
+                $where = ['sender' => $this->modx->user->get('id')];
```
Comment (Mark-H) [inline core/model/modx/processors/security/message/getlist.class.php]: @Alroniks I just noticed this short array syntax on this line, and 51 below. Any chance I could convince you to open a PR that restores PHP 5.3 compatibility?

Sorry for not spotting this sooner!

Comment (Alroniks) [inline core/model/modx/processors/security/message/getlist.class.php]: Fixed https://github.com/modxcms/revolution/pull/13479

**Pull request merged**